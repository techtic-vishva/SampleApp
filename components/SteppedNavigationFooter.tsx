import React from 'react'
import Step from './Step'
import { View } from './Themed'
import OrangeButton from './OrangeButton'
import { GestureResponderEvent, Platform } from 'react-native'

type OnPress = (event: GestureResponderEvent) => void

const vSpace = 30

const buttonStyleBase = {
	marginHorizontal: 20,
	// marginBottom: vSpace,
	flex: 1
}

export default function SteppedNavigationFooter({
	activeStep,
	totalSteps,
	onContinue,
	onBack,
	onSkip,
	onContinueDisabled
}: {
	activeStep: number
	totalSteps?: number
	onContinue?: OnPress
	onBack?: OnPress
	onSkip?: OnPress
	onContinueDisabled?: boolean
}) {
	let buttonCount = 0
	if (onContinue) buttonCount++
	if (onBack) buttonCount++
	if (onSkip) buttonCount++

	const leftStyle = {
		marginRight: buttonCount > 1 ? buttonStyleBase.marginHorizontal / 2 : undefined
	}
	const rightStyle = {
		marginLeft: buttonCount > 1 ? buttonStyleBase.marginHorizontal / 2 : undefined
	}

	return (
		<View style={{ width: '100%', display: 'flex', alignItems: 'center' }}>
			<View style={{ display: 'flex', flexDirection: 'row', marginBottom: vSpace }}>
				{onSkip && (
					<OrangeButton
						onPress={onSkip}
						outterStyle={[buttonStyleBase, { backgroundColor: 'white', borderColor: 'white' }, leftStyle]}
						title="Skip"
					/>
				)}
				{onBack && (
					<OrangeButton
						onPress={onBack}
						outterStyle={[buttonStyleBase, leftStyle]}
						icon="arrow-left"
						title="Back"
						iconLeft={true}
					/>
				)}
				{onContinue && (
					<OrangeButton
						disabled={onContinueDisabled}
						onPress={onContinue}
						outterStyle={[buttonStyleBase, rightStyle]}
						icon="arrow-right"
						title="Continue"
					/>
				)}
			</View>
			{Step(activeStep, totalSteps, {
				marginBottom: Platform.OS === 'ios' ? vSpace + 30 : 20 /* +30 to compensate for swipe bar */
			})}
		</View>
	)
}
