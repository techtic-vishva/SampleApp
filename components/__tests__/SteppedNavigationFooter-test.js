import * as React from 'react'
import renderer from 'react-test-renderer'

import SteppedNavigationFooter from '../SteppedNavigationFooter'

it('render when passing the props', () => {
	const tree = renderer
		.create(
			<SteppedNavigationFooter
				activeStep={1}
				totalSteps={4}
				onContinueDisabled={false}
				onContinue={() => {
					console.log('on continue pressed')
				}}
				onBack={() => {
					console.log('on back pressed')
				}}
				onSkip={() => {
					console.log('on skip pressed')
				}}
			/>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})

it('render when continue button is disable', () => {
	const tree = renderer
		.create(
			<SteppedNavigationFooter
				activeStep={1}
				totalSteps={4}
				onContinueDisabled={true}
				onContinue={() => {
					console.log('on continue pressed')
				}}
				onBack={() => {
					console.log('on back pressed')
				}}
				onSkip={() => {
					console.log('on skip pressed')
				}}
			/>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
