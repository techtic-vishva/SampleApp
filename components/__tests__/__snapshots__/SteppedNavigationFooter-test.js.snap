// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when continue button is disable 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#0b0805",
      },
      Object {
        "alignItems": "center",
        "display": "flex",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#0b0805",
        },
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 30,
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginRight": 10,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "#0b0805",
              "fontSize": 18,
            },
          ]
        }
      >
        Skip
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ed832d",
          "borderColor": "#ed832d",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginRight": 10,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "50%",
        }
      }
    >
      <View
        name="arrow-left"
        size={20}
        style={
          Object {
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "#0b0805",
              "fontSize": 18,
            },
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7c4519",
          "borderColor": "#7c4519",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginLeft": 10,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "#0b0805",
              "fontSize": 18,
            },
          ]
        }
      >
        Continue
      </Text>
      <View
        name="arrow-right"
        size={20}
        style={
          Array [
            Object {
              "paddingHorizontal": 10,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "marginBottom": 60,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 3,
          "marginHorizontal": 5,
          "marginTop": 1,
          "opacity": undefined,
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 5,
          "marginHorizontal": 5,
          "marginTop": 0,
          "opacity": 0.5,
          "width": 5,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 5,
          "marginHorizontal": 5,
          "marginTop": 0,
          "opacity": 0.5,
          "width": 5,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 5,
          "marginHorizontal": 5,
          "marginTop": 0,
          "opacity": 0.5,
          "width": 5,
        }
      }
    />
  </View>
</View>
`;

exports[`render when passing the props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#0b0805",
      },
      Object {
        "alignItems": "center",
        "display": "flex",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#0b0805",
        },
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 30,
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "white",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginRight": 10,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "#0b0805",
              "fontSize": 18,
            },
          ]
        }
      >
        Skip
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ed832d",
          "borderColor": "#ed832d",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginRight": 10,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "50%",
        }
      }
    >
      <View
        name="arrow-left"
        size={20}
        style={
          Object {
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "#0b0805",
              "fontSize": 18,
            },
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ed832d",
          "borderColor": "#ed832d",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginLeft": 10,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "#0b0805",
              "fontSize": 18,
            },
          ]
        }
      >
        Continue
      </Text>
      <View
        name="arrow-right"
        size={20}
        style={
          Array [
            Object {
              "paddingHorizontal": 10,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "marginBottom": 60,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 3,
          "marginHorizontal": 5,
          "marginTop": 1,
          "opacity": undefined,
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 5,
          "marginHorizontal": 5,
          "marginTop": 0,
          "opacity": 0.5,
          "width": 5,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 5,
          "marginHorizontal": 5,
          "marginTop": 0,
          "opacity": 0.5,
          "width": 5,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 5,
          "marginHorizontal": 5,
          "marginTop": 0,
          "opacity": 0.5,
          "width": 5,
        }
      }
    />
  </View>
</View>
`;
